### ISSUE TITLE
Support Thunderbird builds post TB 60

### ISSUE THREAD

**Issue by CyberMan571** [protz/thunderbird-conversations#1308]
* **Thunderbird version:**
Name 	Thunderbird
Version 	64.0a1
Build ID 	20180915120746
* **Thunderbird Conversation version:**
Thunderbird Conversations	2.14.4
* **Operating System:**
OS 	Windows_NT 10.0

The Thunderbird Conversation version of 2.14.4 does not work ing the thunderbird daily build

the progress of the extension in first setup is not continue

like it below

![image](https://user-images.githubusercontent.com/30759335/45592084-bccda380-b96c-11e8-9ad9-81372039259f.png)

**Comment by Standard8** [comment]
This is quite possible, the core of Thunderbird is changing a lot at the moment, and it is highly likely that things will get broken.

When I do get a bit of time, my work is more likely to be for improving the architecture. So if anyone wants to contribute patches, that would be most welcome.

Preferably, patches would support TB 60 as well as the next versions, but there may come a time where we have to break that. 

https://wiki.mozilla.org/Thunderbird/Add-ons_Guide_63 is sometimes a useful resource to reference, though it isn't always updated straight away.

**Comment by scheinercc** [comment]
Just to clarify is this a tracking bug for Daily _and Beta_ now? Thanks

**Comment by Standard8** [comment]
Yes, although #1323 made some improvements for 66, that have landed on master, there's a few regressions that need addressing before we can release it.

The next major version of Thunderbird is not until 68 (which will be around July), so until then, although I'd like to support latest beta if possible, it isn't a priority. Some of it may also be complex, since supporting both 60 + latest may also get complicated from a compatible with both point of view.

As I said before, patches are welcome.

**Comment by Standard8** [comment]
Just to note, the changes that have landed have caused #1346 so we can't release them at the moment until we fix that.

**Comment by Standard8** [comment]
I've started getting master updated to work with 68. The basics work, but quick reply, contact icons and probably a few other things are broken at the moment.

Some of the work to switch to react may benefit the issues around #1346, but switching is quite a long task, so we may need to figure out a workaround to get our images to display in the meantime.

**Comment by Standard8** [comment]
2.15.2 has just been released & approved which now works with Thunderbird 68 betas (nothing earlier).

This has featured some significant rework that is still in progress but helps to fix a number of issues.

Unfortunately at this time I haven't yet reworked quick reply which is completely broken. I have started on it, but found another issue to fix first. Can't yet say if it will be back in time for 68 release or not but hopefully it will.

This release should at least help with testing out the reworked parts and checking for anything broken that I've missed.

**Comment by Standard8** [comment]
If anyone is having an issue after upgrading to TB 68beta (or 69beta), where conversations doesn't seem to replace the normal message format, you can either try removing the add-on and re-installing it, or removing the startup cache file, on Mac it is `~/Library/Caches/Thunderbird/Profiles/<profiledir>/startupCache/startupCache.8.little`. I've filed the cache issue as a Thunderbird issue, so hopefully it'll get solved before [68 final](https://bugzilla.mozilla.org/show_bug.cgi?id=1571453).

**Comment by Standard8** [comment]
The 2.15.x series is receiving reasonably frequent updates, and as far as I know it is working fine in 68. The only significant issue is the quick reply not working, which has been disabled in that series.

I'm now tracking that in #1361.

Hence, I'm closing this out as "fixed", as we're now supporting 68.
